src/systemTest/java/com/flightstats/hub/testcase/WebhookLifecycleTest.java

```diff
@@ -0,0 +1,198 @@
+
+    private static final String EMPTY_STRING = "";
+
+    private ChannelItemResourceClient channelItemResourceClient;
+    private ChannelResourceClient channelResourceClient;
+    private WebhookResourceClient webhookResourceClient;
+    private CallbackResourceClient callbackResourceClient;
+    private CallbackServer callbackServer;
+
+    private String channelName;
+    private String webhookName;
+
+    private List<String> channelItemsPosted;
+
+    @Before
+    public void setup() {
+
+        this.channelItemResourceClient = getHttpClient(ChannelItemResourceClient.class);
+        this.channelResourceClient = getHttpClient(ChannelResourceClient.class);
+        this.webhookResourceClient = getHttpClient(WebhookResourceClient.class);
+        this.callbackResourceClient = getHttpClient(CallbackResourceClient.class);
+        this.callbackServer = injector.getInstance(CallbackServer.class);
+
+        this.callbackServer.start();
+
+        this.channelName = generateRandomString();
+        this.webhookName = generateRandomString();
+    }
+
+    @Test
+    @SneakyThrows
+    public void testWebhookWithNoStartItem() {
+        final String data = "{\"webhook\": \"test1\", \"startitem\":\"false\"}";
+
+        createChannel();
+        addWebhook(buildWebhook(EMPTY_STRING));
+
+        List<String> channelItems = addItemsToChannel(data);
+        verifyWebhookCallback(channelItems);
+    }
+
+    @Test
+    @SneakyThrows
+    public void testWebhookWithStartItem() {
+        final String data = "{\"webhook\": \"test2\", \"startitem\":\"true\"}";
+
+        createChannel();
+        List<String> channelItems = addItemsToChannel(data);
+
+        addWebhook(buildWebhook(channelItems.get(4)));
+        verifyWebhookCallback(channelItems.subList(5, channelItems.size()));
+    }
+
+    @SneakyThrows
+    private void logWebhookCallbackError() {
+        log.info("Webhook name {} ", webhookName);
+
+        Call<Object> call = webhookResourceClient.getError(webhookName);
+        Response<Object> response = call.execute();
+
+        assertEquals(OK.getStatusCode(), response.code());
+        log.info("Call back errors for webhook {} {} ", webhookName, response.body());
+    }
+
+
+    private void verifyWebhookCallback(List<String> channelItems) {
+        Call<String> call = callbackResourceClient.get(callbackServer.getUrl() + "/" + this.webhookName);
+
+        try {
+            await().atMost(90, TimeUnit.SECONDS).until(() -> {
+                Response<String> response = call.clone().execute();
+                channelItemsPosted = parseResponse(response.body());
+                return response.code() == OK.getStatusCode()
+                        && channelItemsPosted.size() == channelItems.size();
+            });
+        } catch (Exception e) {
+            log.error("Problem verifying webhook callbacks. {} ", e.getMessage());
+            logWebhookCallbackError();
+        }
+
+        Collections.sort(channelItems);
```

Comment:
Why does this method sort the items? By doing that, this doesn't verify that the callback receives items in the list they're sent.

Reply:
lists compare works only when 2 lists are sorted.

Reply:
That is odd. I think one of the expectations we have for webhooks is that they'll arrive in order unless there's a failure and retrying is involved. Does the test pass if you sort `channelItems` but not `channelItemsPosted`?